scripts/main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.directive.js

```diff
@@ -2,6 +2,32 @@ angular.module('designer.module')
     .directive('textField', function () {
         return {
             restrict: 'E',
-            templateUrl: 'main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.template.html'
+            templateUrl: 'main/designer/components/inspector/inspector-component/Fields/Text/inspector-text-component.template.html',
+            link: function ($scope) {
+                $scope.showConfigArr = function () {
+                    $scope.finalJSON = [];
+                    debugger;
+                    if ($scope.$root.definitions.components.length > 0) {
+                        for (var i = 0; i < $scope.$root.definitions.components.length; i++) {
+                     /*       if ($scope.$root.definitions.components[i].inspectorConfig.length !== 1) {
+                                for (var j = 0; j < $scope.$root.definitions.components[i].inspectorConfig.length; j++) {
+                                    for (var key in $scope.$root.definitions.components[i].inspectorConfig[j]){
+                                        if (key === "description") {
+                                            var description = $scope.$root.definitions.components[i].inspectorConfig[j][key];
+                                        }
+                                    }
+                                }
+                            }*/
+
+                            $scope.finalJSON.push({
+                                _name: $scope.$root.definitions.components[i]._name,
+                                _type: $scope.$root.definitions.components[i]._type,
+                                label: $scope.$root.definitions.components[i].label,
```

Comment:
`label` and `description` are hard-coded here 

as I mentioned before, try to not relay on prop names

Reply:
also, finalJSON must be in `designer.directive`

Reply:
particular inspector field must have only logic to write value to the model, he must not aware about some finalJSON

responsibility for this filed to accept type and model and render field bused on the type and than write value to the model